modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/StrictWFS_2_0_Strategy.java

```diff
@@ -588,55 +512,230 @@ protected String getOperationURI(WFSOperationType operation, HttpMethod method)
+            break;
+        case GET_FEATURE_WITH_LOCK:
+            parameterName = wfs1_0 ? "ResultFormat" : "outputFormat";
+            break;
+        case TRANSACTION:
+            if (wfs1_0) {
+                //TODO: not sure what to do here.
+                //this is a hack, there appears to be no format info in the 1.0 capabilities for transaction
+                operation = GET_FEATURE;
+                parameterName = "ResultFormat";
+            } else {
+                parameterName = "inputFormat";
             }
-
+            break;
+        default:
+            throw new UnsupportedOperationException("not yet implemented for " + operation);
         }
-        // TODO Auto-generated method stub
-        return null;
+        
+        final OperationType operationMetadata = getOperationMetadata(operation);
+
+        Set<String> serverSupportedFormats;
+        serverSupportedFormats = findParameters(operationMetadata, parameterName);
+        return serverSupportedFormats;
     }
 
     @Override
     public Set<String> getServerSupportedOutputFormats(QName typeName, WFSOperationType operation) {
-        // TODO Auto-generated method stub
-        return null;
+        Set<String> ftypeFormats = new HashSet<String>();
+
+        final Set<String> serviceOutputFormats = getServerSupportedOutputFormats(operation);
+        ftypeFormats.addAll(serviceOutputFormats);
+
+        if (GET_FEATURE.equals(operation)) {
+            final FeatureTypeInfo typeInfo = getFeatureTypeInfo(typeName);
+
+            final Set<String> typeAdvertisedFormats = typeInfo.getOutputFormats();
+
+            ftypeFormats.addAll(typeAdvertisedFormats);
+        }
+        return ftypeFormats;
     }
 
     @Override
     public List<String> getClientSupportedOutputFormats(WFSOperationType operation) {
-        // TODO Auto-generated method stub
-        return null;
+        List<WFSResponseFactory> operationResponseFactories;
+        operationResponseFactories = WFSExtensions.findResponseFactories(operation);
+
+        List<String> outputFormats = new LinkedList<String>();
+        for (WFSResponseFactory factory : operationResponseFactories) {
+            List<String> factoryFormats = factory.getSupportedOutputFormats();
+            outputFormats.addAll(factoryFormats);
+        }
+
+        if (GET_FEATURE.equals(operation)) {
+            for (String preferred : PREFERRED_FORMATS) {
+                boolean hasFormat = outputFormats.remove(preferred);
+                if (hasFormat) {
+                    outputFormats.add(0, preferred);
+                    break;
+                }
+            }
+        }
+
+        return outputFormats;
     }
 
     @Override
     public boolean supportsTransaction(QName typeName) {
-        // TODO Auto-generated method stub
-        return false;
+        try {
+            getFeatureTypeInfo(typeName);
+        } catch (IllegalArgumentException e) {
+            throw e;
+        }
```

Comment:
This is part of the work from last year so my memory isn't that fresh on this code. As I understand it is testing whether
1) the type is actually in the capabilities, otherwise an exception is generated, there is no side effect of parsing because it has already been parsed and it is just checking if it's in memory.
2) transactions itself are supported in the capabilties
3) commented: all three operations are supported. Not sure if this should really happen.

Transactions at the moment are not completely functional, there is still some additional work needed. I suggest I revisit this method then.